### ISSUE TITLE
Next release?

### ISSUE THREAD

**Issue by Symbian9** [openstreetmap/merkaartor#146]
When would be released next version of Merkaartor?

**Comment by KAMiKAZOW** [comment]
I'm using the latest Git snapshot. Working fine. Could be released as-is.

**Comment by Krakonos** [comment]
Hi! Good to know. I will try to find some time to do it, it's long overdue.

**Comment by KAMiKAZOW** [comment]
>  I will try to find some time to do it

Just click on the "Make release" button and call it a beta. Then wait a month a two (for updated translations and bug reports), then click that button again. :)

**Comment by KAMiKAZOW** [comment]
@Krakonos Maybe merge #149 and #151 and make that new release?

**Comment by Symbian9** [comment]
> Krakonos added this to the 0.18.4 milestone

When? When we could see `0.18.4` release?

Latest version in repositories is `0.18.3` that released in 2016...

- https://repology.org/metapackage/merkaartor/versions
![Packaging status](https://repology.org/badge/vertical-allrepos/azpainter.svg)

**Comment by Krakonos** [comment]
I still have some fixing to do. Issue #148 has PR prepared. I have some code for issue #147, but it seems I'm triggering some other problem when Merkaartor is in background. However, I hope to come up with a fix soon. Issue #128 is probably void, as I can not find any place we are generating automatic tags.

TL;DR If everything goes well, we could have a release in a week or two.

**Comment by Krakonos** [comment]
A quick update: I have commited a fix fo #147 as PR #156 . Testing would be appreciated.

**Comment by KAMiKAZOW** [comment]
Did this days ago but somehow forgot to mention it:

Builds for x64 openSUSE Leap 15 and Tumbleweed are available at https://build.opensuse.org/package/show/home:KAMiKAZOW:Test/merkaartor

Install only the package itself, don't add the whole repository.

I could look into Fedora support if there's serious interest.

I'm neither interested in learning .deb packaging (looked into it once: it's horrible) nor bridge technologies like AppImage. I'll adopt Flatpak once OBS supports it.

**Comment by Krakonos** [comment]
I'm currently stuck in #147 and can't figure out the issue. Sometimes I can reproduce the problem, sometimes not and unfortunately, almost any debugging tool hides it. The only useful data I have is gdb backtrace, which points deep into Qt signal handling and so far I've been unable to find the originating object. If someone wants to lend me a hand debugging that code, it would be most appreciated.

**Comment by pabs3** [comment]
FYI, due to the lack of a new release, Merkaartor was removed from Debian:

https://bugs.debian.org/933492

**Comment by Krakonos** [comment]
That's a shame and I apologize to anyone affected by this. However, to make sure this does not happen in the future, I will make myself some time to sort through the bugs and try to get some release as soon as possible.

**Comment by KAMiKAZOW** [comment]
So how about releasing the current state as 0.18.4?

**Comment by Symbian9** [comment]
Would we see `0.18.4` this year?

Or *Merkaartor* development fully abandoned?

**Comment by Krakonos** [comment]
Yes. Actually, I have merged all the work that was ready for the release and I am testing windows/macosx builds. The main problem is that I'm hitting a few issues similar to #182 but slightly worse. I'm not really sure what can be done about it in the short term.

Anyway, if somebody would be willing to test the windows/macosx builds for some basic functionality, that would help a lot.

**Comment by gdt** [comment]
In case anyone cares, merkaator is in danger of being dropped from pkgsrc, for basically the same reasons as Debian.

**Comment by Krakonos** [comment]
Ok. As I currently don't have time to debug the windows issues, I propose to do the release as-is and note the windows issues. Not perfect, but will help a lot of people on linux/macos.

**Comment by Symbian9** [comment]
> As I currently don't have time to debug the windows issues, I propose to do the release as-is and note the windows issues. Not perfect, but will help a lot of people on linux/macos.

Cool! ~~Please, do not release until Nov 27, 2019 for celebrate 3 years of this issue thread!~~ Just do more regular minor releases!

**Comment by gdt** [comment]
I think having a release that works on POSIX-like systems (I'm concerned about NetBSD) and doesn't work on windows is a major improvement over no release.

**Comment by KAMiKAZOW** [comment]
> I propose to do the release as-is and note the windows issues.

Don't propose what I already proposed 1.5 years ago, just click the New Release button already.

**Comment by z3ntu** [comment]
We're getting closer to a 0.18.4 release: https://github.com/openstreetmap/merkaartor/commit/fa61d042a553d60e8522986128aae7b79d5d9952 :)

**Comment by kkofler** [comment]
Fedora also needs the new release: https://bugzilla.redhat.com/show_bug.cgi?id=1736107

**Comment by kkofler** [comment]
https://github.com/openstreetmap/merkaartor/releases/tag/0.18.4 – yay!

**Comment by Krakonos** [comment]
I'm happy to announce that I have tagged the release and uploaded windows & mac osx binaries. I hope it serves well and thanks for the nagging (though I still prefer pull requests!).
